File: 1460-number-of-substrings-containing-all-three-characters/number-of-substrings-containing-all-three-characters.cpp
class Solution {
public:
    int numberOfSubstrings(string s) {
        vector<int> idx(3, 0);
        int count = 0;

        for (int i = 0; i < s.length(); ++i) {
            idx[s[i] - 'a'] = i + 1;

            count += *min_element(idx.begin(), idx.end());
        }

        return count;
    }
};

File: 1460-number-of-substrings-containing-all-three-characters/number-of-substrings-containing-all-three-characters.py
class Solution:
    def numberOfSubstrings(self, s: str) -> int:
        idx = [0] * 3
        count = 0

        for i, c in enumerate(s, 1):
            idx[ord(c) - ord("a")] = i
            count += min(idx)

        return count
